src/components/EventDetails/EventDetails.js

```diff
@@ -18,862 +18,1089 @@
-  }
-  Linking.openURL(url);
-};
-
-const lightningIcon = require('assets/icons/icon_lightning_sm.png');
+const Divider = styled.View`
+  width: 100%;
+  height: 1px;
+  background-color: ${themedColors.tertiary};
+  margin: 8px 0px 18px;
+`;
 
-class EventDetails extends React.Component<Props, State> {
-  timer: ?IntervalID;
-  timeout: ?TimeoutID;
-  // HACK: we need to cache the tx data for smart wallet migration process
-  cachedTxInfo = {};
 
+class EventDetail extends React.Component<Props, State> {
   state = {
-    containerHeight: undefined,
+    isReceiveModalVisible: false,
+    SWActivationModalVisible: false,
   };
 
-  shouldComponentUpdate(nextProps: Props) {
-    return !isEqual(this.props, nextProps);
-  }
+  isReceived = ({ to: address }: Object): boolean => {
+    const { activeAccountAddress, bitcoinAddresses } = this.props;
+    return addressesEqual(address, activeAccountAddress) || bitcoinAddresses.some(e => e.address === address);
+  };
 
-  componentDidMount() {
+  getRelevantAddress = (event: Object): string => {
+    const isReceived = this.isReceived(event);
+    return isReceived ? event.from : event.to;
+  };
+
+  getFeeLabel = () => {
     const {
-      eventType,
-      eventData,
-      updateTransactionStatus,
-      getTxNoteByContact,
-      contacts,
-      history,
-      ensRegistry,
-      lookupAddress,
+      event, baseFiatCurrency, rates, assetDecimals,
     } = this.props;
+    const {
+      gasUsed, gasPrice, btcFee, asset,
+    } = event;
 
-    if (eventType !== TRANSACTION_EVENT) return;
-
-    if (contacts.find(contact => contact.username === eventData.username)
-      && Object.keys(eventData.contact).length) {
-      getTxNoteByContact(eventData.contact.username);
-    }
-
-    const txInfo = history.find(tx => tx.hash === eventData.hash) || {};
-    if (txInfo.status === TX_PENDING_STATUS) {
-      this.timeout = setTimeout(() => updateTransactionStatus(eventData.hash), 500);
-      this.timer = setInterval(() => updateTransactionStatus(eventData.hash), 10000);
-    }
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+    let formattedFee;
 
-    if (txInfo.status === TX_CONFIRMED_STATUS && (!txInfo.gasUsed || !txInfo.gasPrice)) {
-      updateTransactionStatus(eventData.hash);
-    }
-
-    const isReceived = this.wasTransactionReceived(txInfo.to, txInfo.tag);
-    const relatedAddress = isReceived ? txInfo.from : txInfo.to;
-    if (!ensRegistry[relatedAddress]) {
-      lookupAddress(relatedAddress);
+    if (gasUsed) {
+      const fee = gasUsed && gasPrice ? Math.round(gasUsed * gasPrice) : 0;
+      formattedFee = parseFloat(utils.formatEther(fee.toString()));
+    } else {
+      formattedFee = parseFloat(formatUnits(btcFee, assetDecimals));
     }
+    const rate = getRate(rates, asset, fiatCurrency);
+    const formattedFiatValue = formatFiat(formattedFee * rate, fiatCurrency);
+    const feeLabel = `Fee ${formattedFee} ${asset} (${formattedFiatValue})`;
```

Comment:
This is not right - fee (at least for now) is in ETH, not in the asset that was sent.
(And the way its fiat value is calculated is wrong - it should take ETH as an asset to get proper rate)